docs/spec/spec.md

```diff
@@ -0,0 +1,950 @@
+
+```ballerina
+service "RouteGuide" on new grpc:Listener(8980) {
+
+    remote function ListFeatures(RouteGuideFeatureCaller caller, Rectangle rectangle) returns error? {
+
+        foreach Feature feature in FEATURES {
+            if inRange(feature.location, rectangle) {
+                check caller->sendFeature(feature);
+            }
+        }
+    }
+}
+```
+
+## 4.3. Client Streaming RPC
+The RPC service definition of a client streaming call is as follows.
+```proto
+service RouteGuide {
+    rpc RecordRoute(stream Point) returns (RouteSummary) {}
+}
+```
+
+**RPC Using Direct Return**
+
+The Ballerina implementation of the client streaming RPC using a direct return is as follows.
+
+```ballerina
+service "RouteGuide" on new grpc:Listener(8980) {
+
+    remote function RecordRoute(stream<Point, grpc:Error?> clientStream) returns RouteSummary|error {
+        Point? lastPoint = ();
+        int pointCount = 0;
+        int featureCount = 0;
+        int distance = 0;
+
+        decimal startTime = time:monotonicNow();
+        check clientStream.forEach(function(Point p) {
+            pointCount += 1;
+            if pointExistsInFeatures(FEATURES, p) {
+                featureCount += 1;
+            }
+
+            if lastPoint is Point {
+                distance = calculateDistance(<Point>lastPoint, p);
+            }
+            lastPoint = p;
+        });
+        decimal endTime = time:monotonicNow();
+        int elapsedTime = <int>(endTime - startTime);
+        return {point_count: pointCount, feature_count: featureCount, distance: distance, elapsed_time: elapsedTime};
+    }
+}
+```
+
+**RPC Using a Caller**
+
+The Ballerina implementation of the client streaming RPC using a caller return is as follows.
+
+```ballerina
+service "RouteGuide" on new grpc:Listener(8980) {
+
+    remote function RecordRoute(RouteGuideRouteSummaryCaller caller, stream<Point, grpc:Error?> clientStream) returns error? {
+        Point? lastPoint = ();
+        int pointCount = 0;
+        int featureCount = 0;
+        int distance = 0;
+
+        decimal startTime = time:monotonicNow();
+        check clientStream.forEach(function(Point p) {
+            pointCount += 1;
+            if pointExistsInFeatures(FEATURES, p) {
+                featureCount += 1;
+            }
+
+            if lastPoint is Point {
+                distance = calculateDistance(<Point>lastPoint, p);
+            }
+            lastPoint = p;
+        });
+        decimal endTime = time:monotonicNow();
+        int elapsedTime = <int>(endTime - startTime);
+        return caller->sendRouteSummary({point_count: pointCount, feature_count: featureCount, distance: distance, elapsed_time: elapsedTime});
+    }
+}
+```
+
+## 4.4. Bidirectional Streaming RPC
+The RPC service definition of a bidirectional streaming call is as follows.
+```proto
+service RouteGuide {
+    rpc ListFeatures(Rectangle) returns (stream Feature) {}
```

Comment:
```suggestion
    rpc ListFeatures(Rectangle) returns (stream Feature) {};
```

Reply:
The semicolon is not needed https://github.com/grpc/grpc-go/blob/v1.38.0/examples/route_guide/routeguide/route_guide.proto